src/storage/v2/indices/text_index.cpp

```diff
@@ -0,0 +1,470 @@
+                                 .scope_ = label});
+  } catch (const std::exception &e) {
+    throw query::QueryException(fmt::format("Tantivy error: {}", e.what()));
+  }
+  label_to_index_.emplace(label, index_name);
+
+  bool has_schema = false;
+  std::vector<std::pair<PropertyId, std::string>> indexed_properties{};
+  auto &index_context = index_.at(index_name).context_;
+  for (const auto &v : db->Vertices(View::NEW)) {
+    if (!v.HasLabel(View::NEW, label).GetValue()) {
+      continue;
+    }
+
+    if (!has_schema) [[unlikely]] {
+      auto properties = v.Properties(View::NEW).GetValue();
+      for (const auto &[prop_id, prop_val] : properties) {
+        if (prop_val.IsBool() || prop_val.IsInt() || prop_val.IsDouble() || prop_val.IsString()) {
+          indexed_properties.emplace_back(std::pair<PropertyId, std::string>{prop_id, db->PropertyToName(prop_id)});
+        }
+      }
+      has_schema = true;
+    }
+
+    nlohmann::json document = {};
+    nlohmann::json properties = nlohmann::json::value_t::object;
+    for (const auto &[prop_id, prop_name] : indexed_properties) {
+      const auto prop_value = v.GetProperty(View::NEW, prop_id).GetValue();
+      switch (prop_value.type()) {
+        case PropertyValue::Type::Bool:
+          properties[prop_name] = prop_value.ValueBool();
+          break;
+        case PropertyValue::Type::Int:
+          properties[prop_name] = prop_value.ValueInt();
+          break;
+        case PropertyValue::Type::Double:
+          properties[prop_name] = prop_value.ValueDouble();
+          break;
+        case PropertyValue::Type::String:
+          properties[prop_name] = prop_value.ValueString();
+          break;
+        case PropertyValue::Type::Null:
+        case PropertyValue::Type::List:
+        case PropertyValue::Type::Map:
+        case PropertyValue::Type::TemporalData:
+        default:
+          continue;
+      }
+    }
+
+    document["data"] = properties;
+    document["metadata"] = {};
+    document["metadata"]["gid"] = v.Gid().AsInt();
+    document["metadata"]["txid"] = v.impl_.transaction_->start_timestamp;
+    document["metadata"]["deleted"] = false;
+    document["metadata"]["is_node"] = true;
+
+    try {
+      mgcxx::text_search::add_document(
+          index_context,
+          mgcxx::text_search::DocumentInput{
+              .data = document.dump(-1, ' ', false, nlohmann::json::error_handler_t::replace)},
+          kDoSkipCommit);
+    } catch (const std::exception &e) {
+      throw query::QueryException(fmt::format("Tantivy error: {}", e.what()));
+    }
+  }
+
+  // As CREATE TEXT INDEX (...) queries don’t accumulate deltas, db_transactional_accessor_->Commit() does not reach
+  // the code area where changes to indices are committed. To get around that without needing to commit text indices
+  // after every such query, we commit here.
+  try {
+    mgcxx::text_search::commit(index_context);
```

Comment:
What is doCommit flag used for then?

Reply:
Do you mean `kDoSkipCommit`? If so, that’s not a flag, but rather a constant employed to avoid using magic values.
Thanks for the comment, I’ll refactor this method so that the above call of `mgcxx::text_search::add_document` doesn’t skip commit.